src/test/scala/integration/TopicLoaderIntSpec.scala

```diff
@@ -212,6 +212,59 @@ class TopicLoaderIntSpec extends KafkaSpecBase[IO] {
       } yield assertion
     }
 
+    "execute callback if the topic is empty and keep streaming" in withKafkaContext { ctx =>
+      import ctx.given
+
+      val postLoad = records(1 to 15)
+
+      for {
+        loadState  <- Ref.of[IO, Boolean](false)
+        topicState <- Ref.empty[IO, Seq[(String, String)]]
+        _          <- createCustomTopics(NonEmptyList.one(testTopic1))
+        assertion  <- loadAndRunR(NonEmptyList.one(testTopic1))(
+                        _ => loadState.set(true),
+                        r => topicState.getAndUpdate(_ :+ r).void
+                      ).surround {
+                        for {
+                          _         <- eventually(loadState.get.asserting(_ shouldBe true))
+                          _         <- publishStringMessages(testTopic1, postLoad)
+                          assertion <-
+                            eventually(
+                              topicState.get.asserting(_ should contain theSameElementsAs postLoad)
+                            )
+                        } yield assertion
+                      }
+      } yield assertion
+    }
+
+    "execute callback if one topic is empty and keep streaming" in withKafkaContext { ctx =>
+      import ctx.given
+
+      val (forTopic1, forTopic2) = records(1 to 15).splitAt(10)
+      val topics                 = NonEmptyList.of(testTopic1, testTopic2)
+
+      for {
+        loadState  <- Ref.of[IO, Boolean](false)
+        topicState <- Ref.empty[IO, Seq[(String, String)]]
+        _          <- createCustomTopics(topics)
+        _          <- publishStringMessages(testTopic1, forTopic1)
+        assertion  <- loadAndRunR(topics)(
+                        _ => loadState.set(true),
+                        r => topicState.getAndUpdate(_ :+ r).void
+                      ).surround {
+                        for {
```

Comment:
shall we extract this into a val, gets a bit messy with the nested for comps

Reply:
https://github.com/sky-uk/fs2-kafka-topic-loader/pull/45/commits/b6093cd32465deabf02fe36be4c524ab2ffcf6cf